migration-engine/core/src/migration/datamodel_differ/directives.rs

```diff
@@ -0,0 +1,126 @@
+use datamodel::ast;
+use migration_connector::steps::MigrationExpression;
+
+#[derive(Debug)]
+pub(crate) struct DirectiveDiffer<'a> {
+    pub(crate) previous: &'a ast::Directive,
+    pub(crate) next: &'a ast::Directive,
+}
+
+impl<'a> DirectiveDiffer<'a> {
+    fn previous_arguments(&self) -> impl Iterator<Item = &ast::Argument> {
```

Comment:
It would be more readable if this function and the one below would come after the pub functions to allow for top to bottom reading.

Reply:
Addressed in 5ae5b2a56b3264d439af1473a399718bd1a59693